include/types.h

```diff
@@ -0,0 +1,67 @@
+#pragma once
+
+#ifdef USE_NATIVE_F
+//Native type includes
+#ifndef __SIZEOF_INT128__
+#include <boost/multiprecision/cpp_int.hpp>
+#endif
```

Comment:
can you comment which ifdef this endif corresponds to? and similarly for other endifs